cernopendata_client/searcher.py

```diff
@@ -154,12 +154,15 @@ def get_record_as_json(server=None, recid=None, doi=None, title=None):
     record_json = record_api.json()
     if "_files" in record_json["metadata"]:
         del record_json["metadata"]["_files"]
-    if record_json["metadata"]["files"]:
-        for field in record_json["metadata"]["files"]:
-            if "bucket" in field:
-                del field["bucket"]
-            if "version_id" in field:
-                del field["version_id"]
+    try:
```

Comment:
It works OK. However, can you please fix similar problems in the other commands where "flies" may not exist? For example:

```console
$ cernopendata-client get-file-locations --recid 550
...
  File ".../cernopendata_client/searcher.py", line 189, in get_files_list
    for file_ in record_json["metadata"]["files"]:
KeyError: 'files'
```

Generally speaking, instead of code like:

```python
for file_ in record_json["metadata"]["files"]:
    pass
```

it is better to always do:

```python
for file_ in record_json["metadata"].get("files", []):
    pass
```

for all the optional fields.

See `rg -C 1 '\["files"\]'` for all occurrences where this may be needed.

Reply:
P.S. If you do this, then the `try/except` wouldn't be needed, and we could simply do something like:

```python
if record_json["metadata"].get("files", []):
    for field in record_json["metadata"].get("files", []):
        ...
```

so that KeyError wouldn't theoretically mask more situations of not existing keys (files, bucket, version_id) in case the if/for loop code could happen to get longer and longer in the future.

Reply:
Oh, I must have been on a different code, this works now:

```console
$ cernopendata-client get-file-locations --recid 550  --protocol xrootd
```

So if you wish we can merge "as is" without amending the key business...

Reply:
Opened another issue: https://github.com/cernopendata/cernopendata-client/issues/112 for failing python test.